shardeum/archive-server
# BLUE-280: Add Publisher for Transaction Digest to IPFS via Web3.Storage

### Summary
- The **[Transaction DIgest Cron Server](https://github.com/shardeum/archive-server/blob/dev/package.json#L19)** in the Archiver creates Transaction Digests every 10 cycles.
- This PR adds an **`ipfsPublisher`** (as a part of the Transaction DIgest Cron Server) to upload/publish the same Transaction Digest to the IPFS Network via the **[Web3.Storage](https://web3.storage/)** provider.

#### **[Linear Task](https://linear.app/shm/issue/BLUE-280/publishing-transaction-digest-data-to-a-public-network)**

### Merging Notes
It'd be good to merge **[BLUE-283's PR](https://github.com/shardeum/archive-server/pull/89)** into BLUE-280 PR first before merging **`BLUE-280`** to **`dev`**.

#### Testing Notes
1. Before starting with the tests, make sure you register on the web3.storage portal with your email address and subscribe to their free-tier plan (that gives 5 GB of free storage).
2. Access your account on https://console.web3.storage and create a **`Space`**, set any name as per your preference. At the top of the screen you'll see a Root DID string (starts with **`did:key:...`**) of your Space.
3. Copy and paste the same email ID and Root DID of your space to the **[`root_did` and `admin_email` fields in the archiver-config.json](https://github.com/shardeum/archive-server/blob/dev/archiver-config.json#L63-L64)** file respectively and set the **`enableSavingToWeb3Storage`** flag to **`true`**.
4. Run **`npm run txDigestCronServer`** OR **`pm2 start --name txDigestCron npm -- run txDigestCronServer`** to launch the **`txDigestCronServer`** process.
